### PR TITLE
Installs, configures and executes AWS Log agent, passes vars to script

### PR DESCRIPTION
**Pull request by Miserlou**
Related: https://github.com/AlexsLemonade/refinebio/issues/146

## Purpose/Implementation Notes

Simple change: install the logger, configure and run it.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Functional tests

List out the functional tests you've completed to verify your changes work locally.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: infrastructure/instances.tf [hunk 1]
```diff
@@ -76,7 +79,7 @@ data "template_file" "nomad-server-script-smusher" {
 resource "aws_instance" "nomad-server-1" {
   ami = "${data.aws_ami.ubuntu.id}"
   instance_type = "t2.small"
-  availability_zone = "us-east-1a"
+  availability_zone = "${var.region}a"
```
Comment (kurtwheeler) [inline infrastructure/instances.tf]: :+1:

File: infrastructure/nomad-configuration/nomad-client-instance-user-data.tpl.sh [hunk 2]
```diff
@@ -17,6 +17,20 @@
 # Change to home directory of the default user
 cd /home/ubuntu
 
+# Install, configure and launch our CloudWatch Logs agent
+cat <<EOF >awslogs.conf
+[general]
```
Comment (kurtwheeler) [inline infrastructure/nomad-configuration/nomad-client-instance-user-data.tpl.sh]: What's this line do?

File: workers/downloader.nomad.tpl [hunk 3]
```diff
@@ -68,6 +68,15 @@ job "DOWNLOADER" {
 
         volumes = ["${{VOLUME_DIR}}:/home/user/data_store"]
 
+        logging {
+          type = "awslogs"
+          config {
+            awslogs-region = "${{REGION}}",
+            awslogs-group = "data-refinery-log-group-${{USER}}-${{STAGE}}",
+            awslogs-steam = "log-stream-nomad-docker-downloader-${{USER}}-${{STAGE}}"
```
Comment (kurtwheeler) [inline workers/downloader.nomad.tpl]: I'm guessing you want this variable to be `awslogs-stream` not `awslogs-steam`.

File: workers/processor.nomad.tpl [hunk 4]
```diff
@@ -69,6 +69,14 @@ job "PROCESSOR" {
 
         volumes = ["${{VOLUME_DIR}}:/home/user/data_store"]
 
+        logging {
+          type = "awslogs"
+          config {
+            awslogs-region = "${{REGION}}",
+            awslogs-group = "data-refinery-log-group-${{USER}}-${{STAGE}}",
+            awslogs-steam = "log-stream-nomad-docker-processor-${{USER}}-${{STAGE}}"
```
Comment (kurtwheeler) [inline workers/processor.nomad.tpl]: steam -> stream

File: infrastructure/nomad-configuration/nomad-client-instance-user-data.tpl.sh (hunk 2 above)
Comment (Miserlou) [reply infrastructure/nomad-configuration/nomad-client-instance-user-data.tpl.sh]: We need an `awslogs.conf` file to be able to run `./awslogs-agent-setup.py` without any human interaction. This block create that file.

**Pull request merged**